### PR TITLE
decision: mail client

### PR DESCRIPTION
**Pull request by kitzbergerg**
The decision about sending mails from the backend as written in #2.

## Basics

- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: doc/decisions/backend_mail_client.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+[mail_smtp](https://crates.io/crates/mail-smtp) can also be used to send emails (and also requires a mail-server).
+
+## Decision
+
+We should use the crate `lettre` for sending mails.
+
+## Rationale
+
+`lettre` is as of writing this the most widely use crate I have found with 1,343,736 total downloads.
```
Comment (absurd-turtle) [inline doc/decisions/backend_mail_client.md:44]:
```suggestion
`lettre` is as of writing this the most widely used crate I have found with 1,343,736 total downloads.  
```

**Review by absurd-turtle — APPROVED** [review]

**Review by markus2330 — APPROVED** [review]
Great! Can you give an estimation how long it will take to provide an endpoint that can send emails with that lib? (Very easy to very difficult)

**Comment by markus2330** [comment]
Btw. the alternative would be that we post text in a Nextcloud chat or create a Nextcloud Deck card when a user wants to contact us. Probably this is overall easier to be done, as it won't involve an smtp setup. Of course this has the disadvantage that a "reply" button would be missing.

@absurd-turtle what do you think?

**Comment by absurd-turtle** [comment]
> Btw. the alternative would be that we post text in a Nextcloud chat or create a Nextcloud Deck card when a user wants to contact us. Probably this is overall easier to be done, as it won't involve an smtp setup. Of course this has the disadvantage that a "reply" button would be missing.
> 
> @absurd-turtle what do you think?

Yes, it could be done. We would have to send a request to the backend which then makes the request to Nextcloud. The reason is that unauthenticated requests to Nextcloud are difficult to do as far as I read.
I think if we want to receive notifications per mail go directly for mail, if we want to have them in Nextcloud, send them to Nextcloud. I don't think one of them is easier to implement.

**Comment by kitzbergerg** [comment]
> Great! Can you give an estimation how long it will take to provide an endpoint that can send emails with that lib? (Very easy to very difficult)

I think it is about in the middle (a bit more on the easy side). The implementation in Rust shouldn't be that difficult, it is more about coordination and configuration that is difficult (as we need 3 new config env variables, need to decide how to locally test something like this as the mail-server is required and we need access to the mail-server from the prod backend). So like I said it isn't necessarily difficult, but takes some time to think about and plan.

> Btw. the alternative would be that we post text in a Nextcloud chat or create a Nextcloud Deck card when a user wants to contact us. Probably this is overall easier to be done, as it won't involve an smtp setup. Of course this has the disadvantage that a "reply" button would be missing.

What do you mean with reply button? The way I thought about it was: A user has a question and puts it into the form on the frontend (where he also has to enter his email). The backend sends a mail with the question and the users email to somebody (e.g. @markus2330). You send a mail to the email the user entered answering the users question.
Maybe you can do some switcheroo somewhere that the mail appears to be send by the user directly instead of our technical smtp backend user (but still same concept).

**Pull request merged**